totalContributors.py

```diff
@@ -0,0 +1,83 @@
+import subprocess
+import pathlib
+import os.path
+import sys
+from datetime import datetime
+
+
+def add_log(dict, input):
+    for dataset in input:
+        name = dataset.split("\t")[1]
+        value = int(dataset.split("\t")[0])
+        if name in dict:
+            dict[name] += value
+        else:
+            dict[name] = value
+
+
+def retrieve_log(repo, lower_version, upper_version):
+    return subprocess.check_output(
+        ["git", "shortlog", "-s", lower_version + ".." + upper_version],
+        universal_newlines=True,
+        cwd=find_path(repo),
+    ).splitlines()
+
+
+def find_version(time, repo):
+    versions_list = subprocess.check_output(
+        ["git", "tag", "-l"], universal_newlines=True, cwd=find_path(repo)
+    ).splitlines()
+    for version in versions_list:
+        version_time = subprocess.check_output(
+            ["git", "log", "-1", "--format=%ai", version],
+            universal_newlines=True,
+            cwd=find_path(repo),
+        ).split()[0]
+        if datetime.strptime(version_time, "%Y-%m-%d") >= datetime.strptime(time, "%Y-%m-%d"):
```

Comment:
Copy-pasting my previous comment for convenience
> Not sure. If I were to implement this, I'd find a commit hash of the repo that is closest to the timestamp of the webapp/server Zulip repo, instead of returning the closest tag, which is less precise. Need input from @timabbott.

Reply:
What are our steps for moving forward with this PR? Do we just wait for now?

Reply:
Yes, just wait for @timabbott.

Reply:
Hello @ericluoliu  and @rht any feedback on this so far?
I just claimed it and had missed the later part of your conversation.
Let me know if this issue is resolved or still open or pending on @timabbott

Reply:
Yes, I think this issue is resolved. Just waiting for @timabbott to provide some input.

Reply:
Ok @ericluoliu . Thank you for the information and well-done man. Good job!
I will rescind my claim now.